### PR TITLE
fix(typescript): add types to disclosure and idPrefix

### LINKED ISSUE #17700: Add TypeScript types: idPrefix

### LINKED ISSUE #17701: Add TypeScript types: Disclosure

### PR DESCRIPTION
**Pull request by ariellalgilmore**
Closes #17700 
Closes #17701

Add types to Disclosure and idPrefix
#### Changelog

**Changed**

- convert files to `ts`

#### Testing / Reviewing

make sure everything still builds and runs properly

### CODE REVIEW

**Review by emyarod — APPROVED** [review]
looks good to me!

File: packages/react/src/components/IdPrefix/index.tsx [hunk 1]
```diff
@@ -6,10 +6,19 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { PropsWithChildren, ReactNode } from 'react';
 import { IdPrefixContext } from '../../internal/useIdPrefix';
 
-function IdPrefix({ children, prefix }) {
+type IdPrefixProps = {
+  children: ReactNode;
```
Comment (tay1orjones) [inline packages/react/src/components/IdPrefix/index.tsx]:
```suggestion
  children?: ReactNode;
```

File: packages/react/src/components/IdPrefix/index.tsx [hunk 2]
```diff
@@ -6,10 +6,19 @@
...
+
+  /**
+   * The value used to prefix the auto-generated id placed on some DOM elements
+   */
+  prefix: string;
```
Comment (tay1orjones) [inline packages/react/src/components/IdPrefix/index.tsx]:
```suggestion
  prefix?: string;
```

**Review by tay1orjones — APPROVED** [review]
Are these optional?

**Pull request merged**